### PR TITLE
numpy version 1.23.5

### PR DESCRIPTION
**Pull request by ous8292**
Minor version bump from 1.23.4 to 1.23.5
- updated version number
- updated SHA

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
It would be nice to fix test/requires by adding:
```
    # workaround for python/cpython#98706
    - importlib_metadata >=4.13  # [py==311]
```
see https://github.com/conda-forge/numpy-feedstock/blame/main/recipe/meta.yaml#L74-L75

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ outputs:
     script: install_base.bat  # [win]
     build:
       entry_points:
-        - f2py = numpy.f2py.f2py2e:main
+        - f2py = numpy.f2py.f2py2e:main  # [win]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:39]: Does this entry point cause failure on UNIX machines?

Comment (ous8292) [reply recipe/meta.yaml:39]: I believe so. I made this change working on a docker image running it on my local machine. Let me test and make sure it is necessary still.

Comment (ous8292) [reply recipe/meta.yaml]: This caused a failure for the windows machine if I remember correctly. I went ahead and changed it without the `win` selector to double-check and see for a failure.

**Comment by skupr-anaconda** [comment]
I think it's a bug on Prefect:
```
Running command: git -C "C:\b\abs_81dj2ng9a9\clone" remote add origin git@github.com:AnacondaRecipes/numpy-feedstock.git

fatal: detected dubious ownership in repository at '/cygdrive/c/b/abs_81dj2ng9a9/clone'
```

**Review by varlackc — COMMENTED** [review]
The package seems to fail for `win-64` we might have to investigate why this is ocurring. 

we might want to avoid using comparison statements on the `host` section if we can avoid it.

https://github.com/AnacondaRecipes/numpy-feedstock/blob/3a0db7af2c729536912491e136c6ab4b2f47937a/recipe/meta.yaml#L51&L53


https://github.com/AnacondaRecipes/numpy-feedstock/blob/3a0db7af2c729536912491e136c6ab4b2f47937a/recipe/meta.yaml#L83&L85

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**